src/aiida/transports/transport.py

```diff
@@ -802,21 +957,734 @@ def glob0(self, dirname, basename):
+
+    def putfile(self, *args, **kwargs):
+        return self.run_command_blocking(self.putfile_async, *args, **kwargs)
+
+    def puttree(self, *args, **kwargs):
+        return self.run_command_blocking(self.puttree_async, *args, **kwargs)
+
+    def chmod(self, *args, **kwargs):
+        return self.run_command_blocking(self.chmod_async, *args, **kwargs)
+
+    def copy(self, *args, **kwargs):
+        return self.run_command_blocking(self.copy_async, *args, **kwargs)
+
+    def copyfile(self, *args, **kwargs):
+        return self.run_command_blocking(self.copyfile_async, *args, **kwargs)
+
+    def copytree(self, *args, **kwargs):
+        return self.run_command_blocking(self.copytree_async, *args, **kwargs)
+
+    def exec_command_wait(self, *args, **kwargs):
+        return self.run_command_blocking(self.exec_command_wait_async, *args, **kwargs)
+
+    def get_attribute(self, *args, **kwargs):
+        return self.run_command_blocking(self.get_attribute_async, *args, **kwargs)
+
+    def isdir(self, *args, **kwargs):
+        return self.run_command_blocking(self.isdir_async, *args, **kwargs)
+
+    def isfile(self, *args, **kwargs):
+        return self.run_command_blocking(self.isfile_async, *args, **kwargs)
+
+    def listdir(self, *args, **kwargs):
+        return self.run_command_blocking(self.listdir_async, *args, **kwargs)
+
+    def listdir_withattributes(self, *args, **kwargs):
+        return self.run_command_blocking(self.listdir_withattributes_async, *args, **kwargs)
+
+    def makedirs(self, *args, **kwargs):
+        return self.run_command_blocking(self.makedirs_async, *args, **kwargs)
+
+    def mkdir(self, *args, **kwargs):
+        return self.run_command_blocking(self.mkdir_async, *args, **kwargs)
+
+    def remove(self, *args, **kwargs):
+        return self.run_command_blocking(self.remove_async, *args, **kwargs)
+
+    def rename(self, *args, **kwargs):
+        return self.run_command_blocking(self.rename_async, *args, **kwargs)
+
+    def rmdir(self, *args, **kwargs):
+        return self.run_command_blocking(self.rmdir_async, *args, **kwargs)
+
+    def rmtree(self, *args, **kwargs):
+        return self.run_command_blocking(self.rmtree_async, *args, **kwargs)
+
+    def path_exists(self, *args, **kwargs):
+        return self.run_command_blocking(self.path_exists_async, *args, **kwargs)
+
+    def whoami(self, *args, **kwargs):
+        return self.run_command_blocking(self.whoami_async, *args, **kwargs)
+
+    def symlink(self, *args, **kwargs):
+        return self.run_command_blocking(self.symlink_async, *args, **kwargs)
+
+    def glob(self, *args, **kwargs):
+        return self.run_command_blocking(self.glob_async, *args, **kwargs)
+
+    def normalize(self, *args, **kwargs):
+        return self.run_command_blocking(self.normalize_async, *args, **kwargs)
+
+
+# This is here for backwards compatibility
+Transport = BlockingTransport
```

Comment:
I don't know if this makes sense to make blocking the default one, especially if you expose both of them in the API. Shouldn't there be a public class for Blocking and Nonblocking transport which one should use to inherit from?

Reply:
This was just for backward compatibility as Giovanni suggested to call the former blocking `Transport`, now as, `BlockingTransport`